FluxNotes/flux
# Smart on FHIR launch of Compass app

This PR addresses jira 1875. I worked with @dehall on this task since his task is dependent on this one.

- Added code so Compass app (formerly the ccp app) can be launched from smart on FHIR 
- Fixed function calls so that the cancer summary can be loaded. This included deleting cancer specific classes and moving functions into FluxCancerDisorderPresent
- Cancer specific functionality now checks the cancer type first by doing a lookup on the cancer code (added jira 1894 to create a generic way to determine if a code represents a certain type of cancer)
- Because cancer types are now of type FluxCancerDisorderPresent, ER, PR, and HER2 shortcuts were being displayed for non breast cancer conditions so for this PR these shortcuts were disabled. Jira 1899 was created to address the larger issue of adding functionality to filter shortcuts on cancer type


To test:
- Open a browser and go to http://moonshot-dev.mitre.org:4001/?auth_error=&fhir_version_1=r2&fhir_version_2=r2&iss=&launch_ehr=1&launch_url=http%3A%2F%2Flocalhost%3A3000%2Flaunchcompass&patient=&prov_skip_auth=1&provider=&pt_skip_auth=1&public_key=&sb=&sde=&sim_ehr=0&token_lifetime=15&user_pt=
- Enter http://localhost:3000/launchcompass for the launch app url and click the button to launch the app
- Pick any clinician name
- For the patient, pick any patient with a last name of McOde123
- Toggle the condition to a colon cancer and verify that the cancer summary is displayed
- Verify that pilot1, pilot1np, demo1, and demo2 still works and can toggle conditions to view different summaries


_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

**Running the Application**


**Task**


**Documentation**


**Code Quality**


**Tests**



## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code
